### PR TITLE
Feature router

### PR DESCRIPTION
**Pull request by tomaszukovskij**

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -1,54 +1,21 @@
...
   <div>
-    <the-header></the-header>
+    <the-header />
     <main>
-      <carousel-list
-        v-if="carouselListData.length"
-        :carousel-list="carouselListData"
-      ></carousel-list>
-      <product-list :product-list="productList"></product-list>
+      <router-view></router-view>
```
Comment (LukasTyla) [inline src/App.vue]: Nitpick, but doesn't eslint complain about a non-self closing component? 🤔 
```suggestion
      <router-view />
```

File: src/components/carousel/CarouselList.vue [hunk 2]
```diff
@@ -79,6 +79,7 @@ export default {
         : "transform 0.5s ease-in-out");
     },
     moveCarouselHandler(direction) {
+      console.log("click");
```
Comment (LukasTyla) [inline src/components/carousel/CarouselList.vue]:
```suggestion
```

File: src/pages/Home.vue [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<template>
+  <div>
+    <carousel-list
+      v-if="productList.length"
```
Comment (LukasTyla) [inline src/pages/Home.vue:4]: Should probably reference relevant data? 🤔 
```suggestion
      v-if="carouselListData.length"
```

File: vue.config.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+  // devServer: {
+  //   proxy: {
+  //     "/api": {
+  //       target: "http://localhost:3008/",
+  //       changeOrigin: true,
+  //     },
+  //   },
+  // },
```
Comment (LukasTyla) [inline vue.config.js:9]:
```suggestion
```
After these and the respective changes to your api urls the vite build will no longer work correctly.
Consider either changing vue-cli config to match the vite one or simplify vite to no longer handle appended `api` to the start of urls.

File: src/router/index.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import Vue from "vue";
+import VueRouter from "vue-router";
+
+import Home from "../pages/Home.vue";
```
Comment (LukasTyla) [inline src/router/index.js]: Consider having this as a dynamic import as well. 😉

File: src/App.vue (hunk 1 above)
Comment (tomaszukovskij) [reply src/App.vue]: eslint doesnt work, not in my power to setup it correctly :/

File: vue.config.js (hunk 4 above)
Comment (tomaszukovskij) [reply vue.config.js:9]: I'll ditch out vite. Prefer to continue with vue cli.

**Pull request merged**